openid-federation-1_0.xml

```diff
@@ -5752,6 +5752,19 @@ Content-Type: application/json
           <spanx style="verb">client_registration_types_supported</spanx>
           metadata parameter.
 	</t>
+	<t>
+	  Since the RP will have no Client Secret when Automatic Registration
+	  is used, the RP MUST include a
+	  <spanx style="verb">token_endpoint_auth_method</spanx>
+	  parameter in its metadata with a value other than
+	  <spanx style="verb">client_secret_basic</spanx>
```

Comment:
IMO limiting this to `client_secret_basic` is not sufficient. This would allow the RP to use `client_secret_post` or `client_secret_jwt`, which is equally problematic. The RP would need a value that is no `client_secret_*`.
Same applies for the mentioning of `client_secret_basic` a few lines below.

Reply:
OAuth and OIDC have no hard bound on the possible client auth methods.

Thus, let's not try to enumerate all possible client auth methods that aren't compatible with auto reg. In 2014 with OIDC we got 4 methods, in 2020 RFC 8705 added 2 more. The OAuth WG is currently working on a draft-ietf-oauth-attestation-based-client-auth draft.

On second reading, the existing paragraph above (line 5742) looks sufficient.

The RP can approach this in ways that are too many to list here:

* It may be subject to a TA policy that resolves to one specific client auth method.
* The resolved OP metadata may include a single `token_endpoint_auth_method`. 
* Etc.

Reply:
I simplified the treatment, as suggested by @zachmann and @vdzhuvinov.